File: java/selenium/src/test/java/dev/selenium/request/MockResponseTest.java
package dev.selenium.request;

import dev.selenium.TestBase;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
import org.openqa.selenium.By;

public class MockResponseTest extends TestBase {

  @Test
  public void mockResponseBody() {
    //    String mockBody = "<h1>Mock Response</h1>";
    // .  RequestInterceptOptions options =
    //       RequestInterceptOptionsBuilder.addFilter("https://selenium.dev/").build();
    //    driver.network().addRequestHandler(options, req -> req.respond(mockBody));

    driver.get("https://selenium.dev/");
    String h1 = driver.findElement(By.tagName("h1")).getText();
    Assertions.assertEquals("Mock Response", h1, "Does not yet support mocks");
  }
}

File: dotnet/Selenium/Request/MockResponseTest.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using OpenQA.Selenium;

namespace Selenium.Request;

[TestClass]
public class MockResponseTest : TestBase
{
    [TestMethod]
    public async Task MockResponseBody()
    {
        const string mockBody = "<h1>Mock Response</h1>";

        await using var intercept = await Bidi.Network.InterceptRequestAsync(async req =>
        {
            await req.ProvideResponseAsync(new() { Body = mockBody });
        });

        // var options = new RequestInterceptOptions {
        //               UrlPatterns = new List<UrlPattern> { new UrlPattern("https://selenium.dev/") },
        //           };
        // await using var registration = await driver.Network.AddRequestHandlerAsync(
        //     options,
        //     req =>
        //     {
        //         req.Body = mockBody);
        //     });
        
        await Driver.Navigate().GoToUrlAsync("https://selenium.dev/");
        Assert.AreEqual("Mock Response", Driver.FindElement(By.TagName("h1")).Text);
    }
}